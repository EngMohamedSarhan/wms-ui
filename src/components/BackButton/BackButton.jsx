import { ArrowBackIosNew } from "@mui/icons-material";
import { Chip } from "@mui/material";
import PropTypes from "prop-types";
import React from "react";
import { Link } from "react-router-dom";

const BackButton = ({ sx, ...props }) => (
  <Link to="/" data-testid="navigate-home">
    <Chip
      icon={<ArrowBackIosNew fontSize="inherit" />}
      label="Go back to Main List"
      sx={{ ...sx, backgroundColor: "transparent", cursor: "pointer" }}
      {...props}
    />
  </Link>
);

BackButton.propTypes = {
  sx: PropTypes.object,
};

export default BackButton;
