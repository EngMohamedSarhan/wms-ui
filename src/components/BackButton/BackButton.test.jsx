/* eslint-disable no-undef */
import { fireEvent, render } from "@testing-library/react";
import { createMemoryHistory } from "history";
import React from "react";
import { Router } from "react-router-dom";
import ShallowRenderer from "react-shallow-renderer";

import BackButton from "./BackButton";

describe("BackButton Component", () =>
  test("BackButton action", () => {
    const history = createMemoryHistory({ initialEntries: ["/lpn-date-check"] });
    const { getByTestId } = render(
      <Router location={history.location} navigator={history}>
        <BackButton />
      </Router>
    );
    expect(history.location.pathname).toBe("/lpn-date-check");
    const button = getByTestId("navigate-home");
    fireEvent.click(button);
    expect(history.location.pathname).toBe("/");
  }));

describe("BackButton Component", () =>
  test("BackButton snapshot", () => {
    const renderer = new ShallowRenderer();
    const history = createMemoryHistory({ initialEntries: ["/lpn-date-check"] });
    renderer.render(
      <Router location={{ ...history.location, key: "1" }}>
        <BackButton />
      </Router>
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  }));
