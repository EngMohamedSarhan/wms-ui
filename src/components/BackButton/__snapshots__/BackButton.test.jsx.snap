// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton Component BackButton snapshot 1`] = `
<Context.Provider
  value={
    Object {
      "basename": "/",
      "navigator": undefined,
      "static": false,
    }
  }
>
  <Context.Provider
    value={
      Object {
        "location": Object {
          "hash": "",
          "key": "1",
          "pathname": "/lpn-date-check",
          "search": "",
          "state": null,
        },
        "navigationType": "POP",
      }
    }
  >
    <BackButton />
  </Context.Provider>
</Context.Provider>
`;

exports[`BackButton snapshot BackButton snapshot 1`] = `
<Context.Provider
  value={
    Object {
      "basename": "/",
      "navigator": undefined,
      "static": false,
    }
  }
>
  <Context.Provider
    value={
      Object {
        "location": Object {
          "hash": "",
          "key": "1",
          "pathname": "/lpn-date-check",
          "search": "",
          "state": null,
        },
        "navigationType": "POP",
      }
    }
  >
    <BackButton />
  </Context.Provider>
</Context.Provider>
`;
